src/Halogen/Internal/VirtualDOM.purs

```diff
@@ -128,7 +128,19 @@ foreign import vnode
   \  return function(props) {\
   \    return function(children) {\
   \      var VirtualNode = require('virtual-dom/vnode/vnode');\
-  \      return new VirtualNode(name, props, children);\
+  \      var props1 = {\
+  \        attributes: {}\
+  \      };\
+  \      for (var key in props) {\
```

Comment:
For optimum speed it'd still be a for loop, just looping over the array of extracted keys instead. `forEach` is probably plenty fast enough though.